public/Copy-DbaDatabase.ps1

```diff
@@ -707,8 +710,10 @@ function Copy-DbaDatabase {
         if ($InputObject) {
             $Source = $InputObject[0].Parent
             $Database = $InputObject.Name
-        }
-
+        } elseif (-not $Source) {
```

Comment:
This should be checked in the begin block to catch as soon as possible, no point in checking everything else if this is not provided.